### PR TITLE
Bugfixes, part 1

### PR DESCRIPTION
**Pull request by madadam**
First batch of assorted bugfixes: fixes some issues the bootstrap and the stress-test.

### CODE REVIEW

File: src/routing/bootstrap.rs [hunk 1]
```diff
@@ -156,12 +160,19 @@ impl<'a> State<'a> {
...
-                        "Bootstrapping redirected to another set of peers: {:?}",
-                        new_bootstrap_addrs,
-                    );
-                    bootstrap_addrs = new_bootstrap_addrs.to_vec();
+                GetSectionResponse::Redirect(mut new_bootstrap_addrs) => {
+                    // Ignore already used addresses
+                    new_bootstrap_addrs.retain(|addr| !used_addrs.contains(addr));
+
+                    if new_bootstrap_addrs.is_empty() {
+                        debug!("Bootstrapping redirected to the same set of peers we already contacted - ignoring");
```
Comment (maqi) [inline src/routing/bootstrap.rs:168]: is that expecting `mem::take` to throw an error on an empty value of bootstrap_addrs to break the loop ?

File: src/routing/bootstrap.rs [hunk 2]
```diff
@@ -213,7 +230,7 @@ impl<'a> State<'a> {
                             return Ok((response, sender))
                         }
                     }
-                }
+                },
```
Comment (maqi) [inline src/routing/bootstrap.rs:233]: is this added by the cargo-fmt automatically?

File: src/routing/bootstrap.rs [hunk 3]
```diff
@@ -137,11 +137,15 @@ impl<'a> State<'a> {
...
+        // Avoid sending more than one request to the same peer.
+        let mut used_addrs = HashSet::new();
+
         loop {
+            used_addrs.extend(bootstrap_addrs.iter().copied());
             self.send_get_section_request(mem::take(&mut bootstrap_addrs), relocate_details)
                 .await?;
 
-            let (response, sender) = self.receive_get_section_response().await?;
+            let (response, sender) = self.receive_get_section_response(relocate_details).await?;
```
Comment (maqi) [inline src/routing/bootstrap.rs:148]: just to confirm. so this is to resolve a bug previous that:
keeps contact old section to bootstrap even got relocated to new section.
right?

File: src/routing/bootstrap.rs (hunk 1 above)
Comment (madadam) [reply src/routing/bootstrap.rs:168]: No. If `bootstrap_addrs` is empty, we won't send any more requests but we'll still wait for more responses.

File: src/routing/bootstrap.rs (hunk 2 above)
Comment (madadam) [reply src/routing/bootstrap.rs:233]: No. Don't know how it got there. I'll remove it, but unless there are other changes in this PR I'll do it in the next one.

File: src/routing/bootstrap.rs (hunk 3 above)
Comment (madadam) [reply src/routing/bootstrap.rs:148]: The bug was that as a relocated node, we were checking the response against our current name, but we should have been checking it against the destination name instead. So we were incorrectly rejecting valid responses.

File: src/routing/bootstrap.rs (hunk 1 above)
Comment (maqi) [reply src/routing/bootstrap.rs:168]: ok. then there will be high chance that after a `time_out` error, the loop got break, i.e. no more GetSectionResponse come in.
right?

**Review by maqi — DISMISSED** [review]

File: src/routing/bootstrap.rs (hunk 1 above)
Comment (madadam) [reply src/routing/bootstrap.rs:168]: We don't implement any timeout in routing for this, so unless the upper layer adds one, this loops might run forever if we never receive a valid `GetSectionResponse`.

**Pull request merged**